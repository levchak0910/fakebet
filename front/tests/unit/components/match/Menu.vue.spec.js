import {createWrapper, disableFile, cutFromOptions, mapProperties} from "../../__utils__";
import Component from "@/components/match/Menu";

// disableFile();

describe("match/Menu.vue", () => {

    let wrapper, leagueSample;
    let mountWrapper = (options = {}) => {
        let {computed} = cutFromOptions(options, ["computed"]);
        wrapper = createWrapper(Component, {
            computed: {
                leagueList: [],
                selectedLeagues: [],
                ...computed
            },
            ...options,
        });
        leagueSample = wrapper.find("li:nth-of-type(1)");
    };

    it("Testing snapshot", () => {
        mountWrapper({
            computed: {
                leagues: [mapProperties("name", "className"), mapProperties("name", "className")],
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    describe("Testing triggering methods", () => {

        it("selectLeague", () => {
            mountWrapper({
                computed: {
                    leagues: [{name: "League 1", className: "selected"}],
                },
                methods: ["selectLeague"],
            });

            leagueSample.trigger("click");

            expect(wrapper.vm.selectLeague).toBeCalledWith({name: "League 1", className: "selected"});
        });

    });

    describe("Testing computed properties", () => {

        it("leagues", () => {
            mountWrapper({
                computed: {
                    leagueList: ["League 1", "League 2"],
                    selectedLeagues: ["League 2"],
                },
            });

            expect(wrapper.vm.leagues).toEqual([{name: "League 1", className: ""}, {name: "League 2", className: "selected"}]);
        });

    });

    describe("Testing methods", () => {

        it("selectLeague", () => {
            mountWrapper({
                methods: ["_selectLeague"],
            });

            wrapper.vm.selectLeague({name: "League Name"});

            expect(wrapper.vm._selectLeague).toBeCalledWith("League Name");
        });

    });

});
