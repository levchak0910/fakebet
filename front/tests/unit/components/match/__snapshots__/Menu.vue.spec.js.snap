// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match/Menu.vue Testing snapshot 1`] = `
<menu>
  <li
    class="{{className}}"
  >
    {{name}}
  </li>
  <li
    class="{{className}}"
  >
    {{name}}
  </li>
</menu>
`;
